Add life of a ping document. (#8298)
sourcegraph/sourcegraph: doc/dev/architecture/life-of-a-ping.md

# Life of a ping

> Each instance of Sourcegraph periodically performs an [update check]((https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24%40b887ad1+func+check%28+file:updatecheck)) to determine if there is a newer version of Sourcegraph available. This performs an HTTP request to sourcegraph.com with an anonymized site identifier, the current instance version, and a set of anonymized and aggregated statistics about the usage of the instance.

Reviewer: ```suggestion
Each instance of Sourcegraph periodically sends an [HTTP request](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24%40b887ad1+func+check%28+file:updatecheck) to sourcegraph.com to check if an update is available and to send some [anonymized aggregated statistics](https://docs.sourcegraph.com/admin/pings).
```

A Dataflow [job](https://console.cloud.google.com/dataflow/jobs/us-central1/2020-02-05_10_31_47-13247700157778222556?project=telligentsourcegraph&authuser=1) subscribes to the topic, transforms the message payload into a BigQuery [table](https://console.cloud.google.com/bigquery?authuser=1&project=telligentsourcegraph&p=telligentsourcegraph&d=sourcegraph_analytics&t=update_checks&page=table). This job will do a small pre-transform (defined [here](https://console.cloud.google.com/storage/browser/_details/sg-analytics-data/dataflow/pipelines/udf/transform.js?project=telligentsourcegraph&authuser=1)) before being converted into the BigQuery table's schema. These records are inserted into the table in batch.
 

> **Caution**: Data sent into BigQuery must match the table schema **exactly**. You cannot send extra fields or the insertion will fail. Doing this caused a few days worth of data loss between 02/02 and 02/05/2020.

Reviewer: ```suggestion
**Caution**: Data sent into BigQuery must match the table schema **exactly**. You cannot send extra fields or the insertion will fail, which will cause permanent data loss until the issue is fixed.
```
